analyzer/Dockerfile.jooola

```diff
@@ -0,0 +1,39 @@
+FROM python:3-buster as analyzer-build
+
+WORKDIR /build
+
+# How to keep this in sync across the many Dockerfiles ?
+ADD https://raw.githubusercontent.com/libretime/libretime/master/tools/packages.py /tmp/packages.py
+COPY packages.ini /tmp/
+# TODO: Rename packages.ini common section to python
+# TODO: Properly handle python3 setup provided by the initial docker image and
+#       our dependencies automagically installing python from apt.
+#       This mainly has to do the python Gstreamer dependencies.
+RUN python3 /tmp/packages.py \
+    --format=line buster \
+    --exclude=common \
+    /tmp/packages.ini > packages.list
+
+COPY . .
+
+RUN pip wheel --wheel-dir dist --no-deps .
+
+FROM python:3-slim-buster
+
+COPY --from=analyzer-build /build/packages.list /tmp/packages.list
+
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y $(cat /tmp/packages.list) && \
```

Comment:
This needs to be run in the build step for some cases, for example analyzer dependencies requires system packages to build

Reply:
Sorry I don't understand what you mean

Reply:
We need to install system package requirements in the build image for some components, as they are required to install/build pip dependencies

Reply:
If you are thinking about Pycairo, I was able to build the wheel without having the system packages. Or maybe I got lost in the dependency stuff. I 'll need to some research.

Reply:
PyGObject fails to install during build:

```
Building wheels for collected packages: pycairo, PyGObject
  Building wheel for pycairo (setup.py): started                                                         
  Building wheel for pycairo (setup.py): finished with status 'done'
  Created wheel for pycairo: filename=pycairo-1.19.1-cp310-cp310-linux_x86_64.whl size=276674 sha256=30e9797f205a6e877f544c2468e8d40604e4bbdb1f71eb28ea36e349d83590aa
  Stored in directory: /tmp/pip-ephem-wheel-cache-_pqz7v43/wheels/fe/05/de/41f9367a9b16909345795817fbe2a78adee3453126a21e60b1
  Building wheel for PyGObject (PEP 517): started
  Building wheel for PyGObject (PEP 517): finished with status 'error'                        
  ERROR: Command errored out with exit status 1:                                                         
   command: /usr/local/bin/python /usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmp0d29s0ke
       cwd: /tmp/pip-install-23t7nh4o/pygobject_30ac5eccb97542e5a48265165795f8b1                                                                                                                                  
  Complete output (44 lines):                                                                                                                                                                                     
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.10
  creating build/lib.linux-x86_64-3.10/pygtkcompat
  copying pygtkcompat/pygtkcompat.py -> build/lib.linux-x86_64-3.10/pygtkcompat
  copying pygtkcompat/generictreemodel.py -> build/lib.linux-x86_64-3.10/pygtkcompat
  copying pygtkcompat/__init__.py -> build/lib.linux-x86_64-3.10/pygtkcompat
  creating build/lib.linux-x86_64-3.10/gi
  copying gi/types.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/_ossighelper.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/pygtkcompat.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/_gtktemplate.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/_error.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/docstring.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/_constants.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/_propertyhelper.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/_option.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/module.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/__init__.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/_signalhelper.py -> build/lib.linux-x86_64-3.10/gi
  copying gi/importer.py -> build/lib.linux-x86_64-3.10/gi
  creating build/lib.linux-x86_64-3.10/gi/repository 
  copying gi/repository/__init__.py -> build/lib.linux-x86_64-3.10/gi/repository
  creating build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/GdkPixbuf.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/Gdk.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/Gtk.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/GObject.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/Gio.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/GIMarshallingTests.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/GLib.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/keysyms.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/__init__.py -> build/lib.linux-x86_64-3.10/gi/overrides
  copying gi/overrides/Pango.py -> build/lib.linux-x86_64-3.10/gi/overrides
  running build_ext
  Package gobject-introspection-1.0 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `gobject-introspection-1.0.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'gobject-introspection-1.0' found
  Command '('pkg-config', '--print-errors', '--exists', 'gobject-introspection-1.0 >= 1.56.0')' returned non-zero exit status 1.
  
  Try installing it with: 'sudo apt install libgirepository1.0-dev'
  ----------------------------------------
  ERROR: Failed building wheel for PyGObject
Successfully built pycairo
Failed to build PyGObject
ERROR: Could not build wheels for PyGObject which use PEP 517 and cannot be installed directly
WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
The command '/bin/sh -c pip3 install --no-cache-dir -r libretime_analyzer.egg-info/requires.txt' returned a non-zero code: 1
```

Reply:
Have you tried building the wheel with `--no-deps` ?

Reply:
`--no-deps` is not a valid option for `bdist_wheel` command